contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import { ILicenseRegistryV2 } from "../interfaces/registries/ILicenseRegistryV2.sol";
+import { ILicensingModule } from "../interfaces/modules/licensing/ILicensingModule.sol";
+import { IDisputeModule } from "../interfaces/modules/dispute/IDisputeModule.sol";
+import { Errors } from "../lib/Errors.sol";
+import { Licensing } from "../lib/Licensing.sol";
+import { GovernableUpgradeable } from "../governance/GovernableUpgradeable.sol";
+import { ILicenseTemplate } from "contracts/interfaces/modules/licensing/ILicenseTemplate.sol";
+
+/// @title LicenseRegistry aka LNFT
+/// @notice Registry of License NFTs, which represent licenses granted by IP ID licensors to create derivative IPs.
+contract LicenseRegistryV2 is ILicenseRegistryV2, GovernableUpgradeable, UUPSUpgradeable {
+    using Strings for *;
+    using ERC165Checker for address;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @dev Storage of the LicenseRegistry
+    /// @param licensingModule Returns the canonical protocol-wide LicensingModule
+    /// @param disputeModule Returns the canonical protocol-wide DisputeModule
+    /// @custom:storage-location erc7201:story-protocol.LicenseRegistry
+    struct LicenseRegistryStorage {
+        ILicensingModule licensingModule;
+        IDisputeModule disputeModule;
+        address defaultLicenseTemplate;
+        uint256 defaultLicenseConfigId;
+        mapping(address => bool) registeredLicenseTemplates;
+        mapping(address => bool) registeredRoyaltyPolicies;
+        mapping(address => bool) registeredCurrencyTokens;
```

Comment:
This seems redundant, since in the RoyaltyModule we have token whitelists already. To avoid potential inconsistencies from Royalty and Licensing, I think it's safer to use the royalty module to verify if a token is whitelisted.

Alternative options:
- Move the whitelisting logic here, have royalty module check with Licensing.
- Move the whitelisting logic to a separate contract or even Governance, and have both Licensing and Royalty use that list.

Reply:
The idea is to put such protocol-wide configuration into `registries` instead of modules to make our code well structured. I feel it make sense the put such `royalty` and `currency token` register into LicenseRegistry, since they are part of license terms. 

We address this in another PR. 

Removed the two mappings now.